pkg/transport/listener.go

```diff
@@ -92,6 +102,78 @@ func (info TLSInfo) Empty() bool {
 	return info.CertFile == "" && info.KeyFile == ""
 }
 
+func SelfCert(dirpath string, hosts []string) (info TLSInfo, err error) {
```

Comment:
i think we  need to document that this function will write the cert / key to the given dir.